Makefile

```diff
@@ -0,0 +1,363 @@
+	chmod +x $(KN)
+	$(KN) version | grep $(KN_VERSION)
+	ln -sf $(KN) $(LOCAL_BIN)/kn
+.PHONY: kn
+kn: $(KN)
+.PHONY: releases-kn
+releases-kn:
+	$(OPEN) $(KN_RELEASES)
+
+
+
+### TARGETS #
+#
+.DEFAULT_GOAL := help
+
+.PHONY: env
+env:: ## Configure shell env - eval "$(make env)" OR rm .env && make .env && source .env
+	@echo 'unalias m 2>/dev/null || true ; alias m=make'
+.env:
+	$(MAKE) --file $(lastword $(MAKEFILE_LIST)) --no-print-directory env SILENT="1>/dev/null 2>&1" > .env
+
+CURL ?= /usr/bin/curl
+$(CURL):
+	@which $(CURL) \
+	|| ( printf "$(RED)$(BOLD)$(CURL)$(RESET)$(RED) is missing, install $(BOLD)curl$(RESET)\n" ; exit 1)
+.PHONY: curl
+curl: $(CURL)
+
+HELP_TARGET_DEPTH ?= \#\#
+.PHONY: help
+help:
+	@awk -F':+ |$(HELP_TARGET_DEPTH)' '/^[^.][0-9a-zA-Z._%-]+:+.+$(HELP_TARGET_DEPTH).+$$/ { printf "\033[36m%-34s\033[0m %s\n", $$1, $$3 }' $(MAKEFILE_LIST) \
+	| sort
+
+define MAKE_TARGETS
+  awk -F':+' '/^[^.%\t_][0-9a-zA-Z._%-]*:+.*$$/ { printf "%s\n", $$1 }' $(MAKEFILE_LIST)
+endef
+
+define BASH_AUTOCOMPLETE
+  complete -W \"$$($(MAKE_TARGETS) | sort | uniq)\" make gmake m
+endef
+
+.PHONY: bash-autocomplete
+bash-autocomplete:
+	@echo "$(BASH_AUTOCOMPLETE)"
+env:: bash-autocomplete
+
+.PHONY: dep-update
+dep-update: ## Update any dependency
+	@printf "Update dep in go.mod by running e.g. $(BOLD)go get -d github.com/rabbitmq/messaging-topology-operator@v1.2.1$(RESET)\n" \
+	; read -rp " (press any key when done)" -n 1
+	$(CURDIR)/hack/update-deps.sh
+	$(CURDIR)/hack/update-codegen.sh
+
+.PHONY: test-unit
+test-unit: ## Run unit tests
+	@printf "$(INFO)Starting point: $(BOLD).github/workflows/knative-go-test.yaml$(RESET)\n"
+	go test -race $(GOTEST) ./... \
+	| grep -v "no test files"
+
+.PHONY: test-unit-uncached
+test-unit-uncached: GOTEST = -count=1
+test-unit-uncached: test-unit
+
+SYSTEM_NAMESPACE = knative-eventing
+export SYSTEM_NAMESPACE
+RABBITMQ_SYSTEM_NAMESPACE = rabbitmq-system
+export RABBITMQ_SYSTEM_NAMESPACE
+RABBITMQ_SOURCE_NAMESPACE = knative-sources
+export RABBITMQ_SOURCE_NAMESPACE
+CERT_MANAGER_NAMESPACE = cert-manager
+export CERT_MANAGER_NAMESPACE
+KIND_CLUSTER_NAME = eventing-rabbitmq-e2e
+export KIND_CLUSTER_NAME
+env::
+	@echo 'export KIND_CLUSTER_NAME="$(KIND_CLUSTER_NAME)"'
+KO_DOCKER_REPO := kind.local
+env::
+	@echo 'export KO_DOCKER_REPO="$(KO_DOCKER_REPO)"'
+export KO_DOCKER_REPO
+MIN_SUPPORTED_K8s_VERSION := 1.20
+KIND_K8s_VERSION = $(MIN_SUPPORTED_K8s_VERSION)
+export KIND_K8s_VERSION
+# Find the corresponding version digest in https://github.com/kubernetes-sigs/kind/releases
+KIND_K8s_DIGEST := sha256:cbeaf907fc78ac97ce7b625e4bf0de16e3ea725daf6b04f930bd14c67c671ff9
+export KIND_K8s_DIGEST
+
+.PHONY: kind-cluster
+kind-cluster: | $(KIND) $(ENVSUBST)
+	( $(KIND) get clusters | grep $(KIND_CLUSTER_NAME) ) \
+	|| ( cat $(CURDIR)/test/e2e/kind.yaml \
+	     | $(ENVSUBST_SAFE) \
+	     | $(KIND) create cluster --name $(KIND_CLUSTER_NAME) --config - )
+
+$(KUBECONFIG): | kind-cluster $(KUBECONFIG_DIR)
```

Comment:
I don't think we should make everything dependent on `kind-cluster`, it means they aren't reusable against an already-existing cluster (GKE, a differently named kind cluster, etc). They should just assume a valid kubernetes context has been set.

Reply:
Fixed in https://github.com/knative-sandbox/eventing-rabbitmq/pull/525/commits/0a4972f9b9330b190643cb50df1b2ac575a8aacc